components/TButtonShare.vue

```diff
@@ -39,11 +39,17 @@ import { useAuth } from '~/use/auth'
 export default {
   setup() {
     const { currentCity } = useCities()
-    const { uid } = useAuth()
+    const {
+      uid,
+      account: {
```

Comment:
you need to import only ref variable, in this case it's account, then you can expose it and use in other methods